### ISSUE TITLE
Standard remote snapshot labels?

### ISSUE THREAD

**Issue by jonjohnsonjr** [containerd/stargz-snapshotter#326]
The labels [here](https://github.com/containerd/stargz-snapshotter/blob/eae95630f9e5ac6e71a7afe7058b349fcf166240/fs/source/source.go#L63-L73) seem to serve the same purpose as [these](https://github.com/containerd/containerd/blob/e47400cbd2d5f45ab8d836d56c8521f6c81d0ce2/pkg/cri/server/image_pull.go#L509-L523) other than the `TargetPrefetchSizeLabel`.

Is there a reason for those three label keys to diverge? Could they be reused instead?

I was digging through `nerdctl` and came across [this](https://github.com/containerd/nerdctl/blob/b0b550fcafb80876be27dde28daa00cc242c2f32/pkg/imgutil/imgutil.go#L153-L160), which has some hard-coded logic for stargz. I'm wondering if it's possible to agree on some standard configuration for determining whether or not something is a remote snapshotter rather than looking specifically for stargz there... what do you think?

**Comment by ktock** [comment]
> The labels here seem to serve the same purpose as these other than the TargetPrefetchSizeLabel.
> Is there a reason for those three label keys to diverge? Could they be reused instead?

Yes, these label keys are diverse but have the same purpose. For tools other than CRI plugin (e.g. BuildKit, nerdctl, etc), [default labels](https://github.com/containerd/stargz-snapshotter/blob/eae95630f9e5ac6e71a7afe7058b349fcf166240/fs/source/source.go#L63-L73) should be used.

> I was digging through nerdctl and came across this, which has some hard-coded logic for stargz. I'm wondering if it's possible to agree on some standard configuration for determining whether or not something is a remote snapshotter rather than looking specifically for stargz there... what do you think?

Isn't referring snapshotter name enough? What's your use-case for this?


Stargz-related configurations are hard-coded to some of stargz-aware tools as of now and I agree that we should make them configurable (especially for making it easier to integrate other types of (non-stargz) remote snapshotters).
